provider/ansible/gcp_utils.py

```diff
@@ -133,11 +133,11 @@ def _validate(self):
     def _credentials(self):
         cred_type = self.module.params['auth_kind']
         if cred_type == 'application':
-            credentials, project_id = google.auth.default()
+            credentials, project_id = google.auth.default(scopes=self.module.params['scopes'])
             return credentials
         elif cred_type == 'serviceaccount':
             path = os.path.realpath(os.path.expanduser(self.module.params['service_account_file']))
-            return service_account.Credentials.from_service_account_file(path)
+            return service_account.Credentials.from_service_account_file(path).with_scopes(self.module.params['scopes'])
         elif cred_type == 'machineaccount':
```

Comment:
GCE machines get scopes when they are created and they can't be changed.

From https://google-auth.readthedocs.io/en/latest/reference/google.auth.compute_engine.html#module-google.auth.compute_engine

```
Compute Engine instances can be created with scopes and therefore these credentials are considered to be ‘scoped’. However, you can not use with_scopes() because it is not possible to change the scopes that the instance has.
```